packages/core/src/service/Downloader.ts

```diff
@@ -31,11 +35,14 @@ export class Downloader {
 	): Promise<R | undefined> {
 		const { id, cache, uri, options, transformer, ttl } = job
 		if (ttl && this.#memoryCache.has(uri)) {
-			const { buffer, time } = this.#memoryCache.get(uri)!
-			if (time <= performance.now() + ttl) {
+			const memoryCacheEntry = this.#memoryCache.get(uri)!
```

Comment:
annoying that we need the non-null assertion here. I see that it can't be undefined because of the `has` above

Reply:
Yeah bit of a typescript quirk, but not that important. We have this pattern in a couple of other places.